flexget/components/sites/sites/serienjunkies.py

```diff
@@ -27,10 +28,80 @@
 regex_is_subtitle = re.compile(r'Untertitel|Subs?|UT', re.I)
 
 LANGUAGE = ['german', 'foreign', 'subtitle', 'dual']
-HOSTER = ['ul', 'cz', 'so', 'all']
+HOSTER = ['ul', 'so', '1fichier', 'ddl', 'filefactory', 'filer', 'nitroflare', 'oboom', 'rapidgator', 'turbobit',
+          'uploaded', 'share-online', 'all']
 
 DEFAULT_LANGUAGE = 'dual'
-DEFAULT_HOSTER = 'ul'
+DEFAULT_HOSTER = 'uploaded'
+
+HOSTER_MAP = {
+    '1fichier': ['1fichier.com'],
+    'ddl': ['ddl.to'],
+    'filefactory': ['filefactory.com'],
+    'filer': ['filer.net'],
+    'nitroflare': ['nitroflare.com'],
+    'oboom': ['oboom.com'],
+    'rapidgator': ['rapidgator.net'],
+    'turbobit': ['turbobit.net'],
+    'uploaded': ['uploaded.to', 'uploaded.net'],
+    'share-online': ['share-online.to']
+}
+
+
+class SerienjunkiesMatch:
+    def __init__(self):
+        self.hoster = None
+        self.quality = float("inf")
+        self.found = False
+
+    @classmethod
+    def factory(cls, href, next_sibling, hoster_config):
+        instance = cls()
```

Comment:
if the first thing you do is init the class then there is no reason to make this a classmethod, you can just move this logic to the init (or keep it as a separate function that init calls)